restdocs-api-spec-openapi-generator/src/test/kotlin/com/epages/restdocs/apispec/openapi2/OpenApi20GeneratorTest.kt

```diff
@@ -168,6 +170,43 @@ class OpenApi20GeneratorTest {
         thenValidateOpenApi(openapi)
     }
 
+    @Test
+    fun `should generate different schema names for different schema attributes`(){
+        //given
+        val ordersFieldDescriptors =  givenFieldDescriptors("_embedded.orders[]")
+        val shopsFieldDescriptors =  givenFieldDescriptors("_embedded.shops[]")
+
+        val ordersSchema: Model  = givenModel(ordersFieldDescriptors)
+        val shopsSchema: Model  = givenModel(shopsFieldDescriptors)
+
+        val schemaNameAndSchemaMap = HashMap<String, Model>()
+
+        //when
+        whenExtractOrFindSchema(schemaNameAndSchemaMap, ordersSchema, shopsSchema)
+
+        //then
+        then(schemaNameAndSchemaMap.size).isEqualTo(2)
```

Comment:
Great! This test would even stay green if the equals method on ObjectProperty would work correctly and we would move back to the aggregation of equal schemas! 👍